### PR TITLE
io-util, coredump: introduce iovw_append() and use it

### PR DESCRIPTION
**Pull request by yuwata**
Follow-up for 946dc7c635f050129896d1515c08a81504af2421.

### CODE REVIEW

File: src/basic/io-util.h [hunk 1]
```diff
@@ -106,10 +106,12 @@ static inline int iovw_consume(struct iovec_wrapper *iovw, void *data, size_t le
                 free(data);
         return r;
 }
+int iovw_put_dup(struct iovec_wrapper *iovw, const void *data, size_t len);
```
Comment (bluca) [inline src/basic/io-util.h]: why is this added in this commit? doesn't seem used

Comment (yuwata) [reply src/basic/io-util.h]: Ouch, I forgot to drop it.

**Review by bluca — APPROVED** [review]
